js/events/click.js

```diff
@@ -96,6 +97,29 @@ const useNativeClick =
     let prevented = null;
     let lastFiredEvent = null;
 
+    function onNodeRemove() {
+        lastFiredEvent = null;
+    }
+
+    const eventNodesDisposing = (() => {
+        function nodesByEvent(event) {
+            return event && [
+                event.target,
+                event.delegateTarget,
+                event.relativeTarget
+            ].filter(node => !!node);
+        }
+
+        return {
+            subscribe: (event, callback) => {
+                eventsEngine.on(nodesByEvent(event), REMOVE_EVENT_NAME, callback);
+            },
+            unsubscribe: (event, callback) => {
+                eventsEngine.off(nodesByEvent(event), REMOVE_EVENT_NAME, callback);
+            }
+        };
+    })();
```

Comment:
I suggest using static class methods or two pure functions instead of make a such strange and complex structure.
The main points are
- get rid of auto called function; (it's our performance...)
- get rid of excess closure.